src/inthandler.S

```diff
@@ -11,134 +11,163 @@
 inthandler:
 	.global inthandler
 
-	/*Save $2 before using*/
-	sd $2,save02
-
 	.set noat
-	/*Fetch exception pc off cop#0*/
-	mfc0 $2,C0_EPC
-
-	la k1,save01
-	sd $1,(k1)
+	.set noreorder
 
-	mfc0 k1,C0_SR
-	la $1,saveSR
-	sw k1,($1)
-	la $1, ~1
-	and k1,$1
-	mtc0 k1,C0_SR
-	.set at
+#define EXCEPTION_CODE_SYS_CALL      (8<<2)
+#define EXCEPTION_CODE_BREAKPOINT    (9<<2)
+#define EXC_STACK_SIZE (544+32)
```

Comment:
A comment on the size would be nice IMHO

Reply:
Done. I've also added a static assert that `sizeof(reg_block_t) == 544`.